add instruction on enum parameter in "getting started" (#12578)
argoproj/argo-workflows: docs/walk-through/parameters.md

```

In this workflow, both steps `A` and `B` would have the same log-level set to `INFO` and can easily be changed between workflow submissions using the `-p` flag.

> 

Reviewer: ```suggestion

# `enum` support

> v3.0 and after

```

Let's put this in a new section since it is distinct from the rest.

We can also add a version specifier for consistency with the rest of the docs. #4314 was released as part of [v2.12](https://github.com/argoproj/argo-workflows/blob/v3.5.4/CHANGELOG.md#v2120-rc1-2020-11-06), while #4365 was released as part of [v3.0](https://github.com/argoproj/argo-workflows/blob/v3.5.4/CHANGELOG.md#v300-rc1-2021-02-08)

> If using the UI, the `enum` parameter type may also be used to restrict the parameter to certain values.

Reviewer: ```suggestion
In the UI, the `enum` field can be used to restrict a parameter to certain values.
```

- more simple and direct, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
- terminology: it's a "field", not a "parameter type"
- grammar: "a" parameter, not specific to "the"

Reviewer: ```suggestion
With `WorkflowTemplates`, you [can use](../workflow-templates.md#UI) the `enum` field to restrict a parameter to certain values in the UI.
```

After realizing https://github.com/argoproj/argo-workflows/pull/12578#discussion_r1468235830, I think it might be better to remove the example entirely from here and just link to the existing example in the `WorkflowTemplate` docs as that is advanced usage (the walk through is mostly for beginners to / new users of Workflows).

If you'd like, you can also improve that example as part of this PR (e.g. adding versions, improving the wording, perhaps moving it out of its current section)

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: hello-world-enum-parameters-
spec:
  entrypoint: whalesay
  arguments:
    parameters:
    - name: message
      default: hello

>       enum:

Reviewer: huh I've actually never used this as a top-level Workflow parameter 👀  this use-case is indeed undocumented currently then... so I guess we should document it here as there's no other doc on parameters

Reviewer: Oh, this _is_ documented elsewhere; #7114 added this under the `WorkflowTemplate` page. That partly makes sense since top-level parameters would only be available with `WorkflowTemplates` -- if you submit a Workflow directly, you wouldn't have access to the UI selection menu. The walk through actually doesn't mention `WorkflowTemplates` at all, so back to square one of me thinking this doesn't quite fit here

I think one of the problems with the current placement is that it's under "Managing WorkflowTemplates" but this feature is not specific to maintaining them, it is for using them. I see that there's no real other place for `WorkflowTemplate` docs currently though.

Maybe we just link to those docs instead then

Author: I was just looking back at how I used `enum` it was in a workflow template, so this example is definitely wrong.

I do think I would have found the docs if they were in the UI section. We could move [this section ](https://github.com/Hadrian-MTV/charts/blob/main/applications/flow/factory-execution/values-prod.yaml#L12)to `UI Features` and add screenshots? And then reference the new page within the `WorkflowTemplate` page.

Reviewer: > We could move [this section](https://github.com/Hadrian-MTV/charts/blob/main/applications/flow/factory-execution/values-prod.yaml#L12)

Your link above 404s for me; that seems to be a link to a private repo's Helm values?

> to `UI Features` and add screenshots?

The [Intermediate Parameters](https://argo-workflows.readthedocs.io/en/release-3.5/intermediate-inputs/) page is already under UI Features. Usage with WorkflowTemplates could perhaps be a prelude to that all on one page

Author: Sorry! [This](https://argo-workflows.readthedocs.io/en/latest/workflow-templates/#ui) is where I meant to link to.

And that makes sense to me.

Reviewer: Ok I thought that was what you meant 😅.
I think we can move that to the Intermediate Parameters page and rename the title there a bit

      - hello
      - goodbye

  templates:
  - name: whalesay
    inputs:
      parameters:
      - name: message       # parameter declaration
    container:
      # run cowsay with that message input parameter as args
      image: docker/whalesay
      command: [cowsay]
      args: ["{{inputs.parameters.message}}"]
```

> `enum` may also be used to create [Intermediate Parameters](../intermediate-inputs.md).

Reviewer: ```suggestion
The `enum` field can also be used with [Intermediate Parameters](../intermediate-inputs.md).
```

- consistency: refer to `enum` the same way, use "can"
- I'm actually not sure that `enum` is necessarily required for that feature; `valueFrom.supplied` is the main thing I think. In general, it could use some more details. so let's just say it can be used with that feature
